packages/desktop-client/src/components/autocomplete/Autocomplete.js

```diff
@@ -454,26 +445,48 @@ function defaultItemToString(item) {
   return item ? getItemName(item) : '';
 }
 
-function _SingleAutocomplete({
-  props: {
+function SingleAutocomplete(props) {
+  const [curState, setState] = React.useState(() => getInitialState({ props }));
+  const { value, selectedItem, filteredSuggestions, highlightedIndex, isOpen } =
+    curState;
+
+  const prevProps = usePrevious(props);
+  React.useEffect(() => {
+    if (!prevProps) return;
+
+    const newState = componentWillReceiveProps(
+      { props: prevProps, state: curState },
+      props,
+    );
+
+    if (newState) {
+      setState(Object.assign({}, curState, newState));
+    }
+  }, [props, prevProps]);
+
+  const updater =
+    operation =>
+    (...arg) => {
+      const newState = Object.assign(
+        {},
+        curState,
+        operation({ props, state: curState }, ...arg) || {},
+      );
+      setState(newState);
+    };
```

Comment:
This is basically what `lively` does. I dislike the new implementation and I plan to return to it very soon. For now: doing minimal changes in order to remove `lively`.